src/apps/filtering.js

```diff
@@ -29,27 +37,44 @@ const searchFilter = (query?: string) => ({ name, currencyId }) => {
 };
 
 const typeFilter = (
-  type = "all",
+  type: AppType[] = ["all"],
   updateAwareInstalledApps: UpdateAwareInstalledApps
 ) => app => {
-  switch (type) {
-    case "installed":
-      return updateAwareInstalledApps.hasOwnProperty(app.name);
-    case "not_installed":
-      return !updateAwareInstalledApps.hasOwnProperty(app.name);
-    case "updatable":
-      return (
-        updateAwareInstalledApps.hasOwnProperty(app.name) &&
-        !updateAwareInstalledApps[app.name]
-      );
-    case "supported":
-      return (
-        app.currencyId &&
-        isCurrencySupported(getCryptoCurrencyById(app.currencyId))
-      );
-    default:
-      return true;
-  }
+  let validApp = true;
+
+  type.forEach(filter => {
```

Comment:
logic can be simplified a bit

- `type` naming is a bit weird, can be just "filters"
- use of `.every` would simplify the logic and not requires to have a `validApp` variable.

```js
const typeFilter = (
  filters: AppType[] = ["all"],
  updateAwareInstalledApps: UpdateAwareInstalledApps
) => app => filters.every(filter => {
  switch (filter) {
    case "installed": return !!updateAwareInstalledApps.hasOwnProperty(app.name);
    ...
  }
})`

Reply:
noice

Reply:
You have it like you asked for now